test/modules/http/compliance/http-res-destroy.test.ts

```diff
@@ -12,57 +14,158 @@ const httpServer = new HttpServer((app) => {
 const interceptor = new ClientRequestInterceptor()
 
 beforeAll(async () => {
-  interceptor.apply()
   await httpServer.listen()
 })
 
 afterEach(() => {
   interceptor.removeAllListeners()
+  interceptor.dispose()
 })
 
 afterAll(async () => {
-  interceptor.dispose()
   await httpServer.close()
 })
 
-it('emits the "error" event when a bypassed response is destroyed', async () => {
-  const socketErrorListener = vi.fn()
+const scenarios = [
+  {
+    condition: 'when the interceptor is not enabled [baseline test]',
+    setup: () => {},
+  },
+  {
+    condition: 'for a bypassed response',
+    setup: () => {
+      interceptor.apply()
+    },
+  },
+  {
+    condition: 'for a mocked response',
+    setup: () => {
+      interceptor.apply()
+      interceptor.on('request', ({ controller }) => {
+        controller.respondWith(new Response('hello world'))
+      })
+    },
+  },
+]
 
-  const request = http
-    .get(httpServer.http.url('/'))
-    .on('socket', (socket) => {
-      socket.on('error', socketErrorListener)
-    })
-    .on('response', (response) => {
-      response.destroy(new Error('reason'))
-    })
+scenarios.forEach(({ condition, setup }) => {
+  it(`emits the "error" event to the socket, response and request when the response is destroyed ${condition}`, async () => {
+    setup()
 
-  const { res } = await waitForClientRequest(request)
+    const socketErrorListener = vi.fn()
+    const responseErrorListener = vi.fn()
+    const requestErrorListener = vi.fn()
 
-  expect(res.destroyed).toBe(true)
-  expect(socketErrorListener).toHaveBeenCalledOnce()
-  expect(socketErrorListener).toHaveBeenCalledWith(new Error('reason'))
-})
+    let socket: Socket
+    const responseError = new Error('reason')
+
+    const request = http
+      .get(httpServer.http.url('/'))
+      .on('socket', (newSocket) => {
+        newSocket.on('error', socketErrorListener)
+        socket = newSocket
+      })
+      .on('response', (response) => {
+        response.on('error', responseErrorListener)
+        response.destroy(responseError)
+      })
+      .on('error', requestErrorListener)
+
+    const { res } = await waitForClientRequest(request)
+
+    await setTimeout(0)
```

Comment:
Some of the event handling etc. is performed async so we have to yield the event loop.